onnxmltools/utils/metadata_props.py

```diff
@@ -0,0 +1,57 @@
+import warnings
+from ..convert.common.case_insensitive_dict import CaseInsensitiveDict
+from ..proto import onnx, onnx_proto
+from distutils.version import StrictVersion
+
+
+KNOWN_METADATA_PROPS = CaseInsensitiveDict({
+    'Image.BitmapPixelFormat': ['gray8', 'rgb8', 'bgr8', 'rgba8', 'bgra8'],
+    'Image.ColorSpaceGamma': ['linear', 'srgb'],
+    'Image.NominalPixelRange': ['nominalrange_0_255', 'normalized_0_1', 'normalized_1_1', 'nominalrange_16_235'],
+})
+
+
+def _validate_metadata(metadata_props):
+    '''
+    Validate metadata properties and possibly show warnings or throw exceptions.
+
+    :param metadata_props: A dictionary of metadata properties, with property names and values
+    '''
+    if len(CaseInsensitiveDict(metadata_props)) != len(metadata_props):
+        raise RuntimeError('Duplicate metadata props found')
+
+    for key, value in metadata_props.items():
+        valid_values = KNOWN_METADATA_PROPS.get(key)
+        if valid_values and value.lower() not in valid_values:
+            warnings.warn('Key {} has invalid value {}. Valid values are {}'.format(key, value, valid_values))
+
+
+def add_metadata_props(onnx_model, metadata_props, targeted_onnx=onnx.__version__):
```

Comment:
Please add doc string and describe what types of the inputs. For metadata_props, you can add a reference to the ONNX page which defines allowed key-value pairs. It'd be nice if you can just copy one or two pairs from that page and then use them as an example here.